server/src/test/java/org/elasticsearch/common/unit/ProcessorsTests.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.unit;
+
+import org.elasticsearch.test.ESTestCase;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.lessThan;
+import static org.hamcrest.Matchers.lessThanOrEqualTo;
+
+public class ProcessorsTests extends ESTestCase {
+    public void testTruncatesAfterFiveDecimalPlaces() {
+        final double processorCount = randomNumberOfProcessors();
+
+        final Processors processors = Processors.of(processorCount);
+        final String processorsString = Double.toString(processors.count());
+        final int decimalPlaces;
+        if (processorsString.contains("E")) {
+            int exponent = Integer.parseInt(processorsString.substring(processorsString.indexOf("E") + 1));
+            assertThat(exponent, is(lessThan(0)));
+            decimalPlaces = Math.abs(exponent);
+        } else {
+            decimalPlaces = processorsString.length() - processorsString.indexOf(".") - 1;
+        }
+        assertThat(decimalPlaces, is(lessThanOrEqualTo(Processors.NUMBER_OF_DECIMAL_PLACES)));
+    }
+
+    public void testRounding() {
+        {
```

Comment:
Can we add 1-2 more cases:

1. A number higher than 1.something, for instance 10.1.
2. A number that is already an integer, for instance 10.